src/thresholds.service.js

```diff
@@ -103,44 +28,51 @@ class ThresholdsService {
       return
     }
 
-    const { weeklyLevels, weeksOfStock, targetPopulations } = getFactors(stockCount, location)
+    const transformedLocation = transform(location, productCoefficients, stockCount.date)
```

Comment:
Nit: `transformedLocation` to me reads like it'll be a copy of the full location, with some modified properties, whereas, `transform` returns a subset of props. Maybe `transformedLocation` -> `factors` or `locationFactors`?